### PR TITLE
12.2.0 Changelog

### PR DESCRIPTION
**Pull request by lilgreenbird**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,21 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 
+## [12.2.0] Preview Release
+### Added
+
+### Changed
+- Renamed driver DefaultAzureCredential authentication to ActiveDirectoryDefault [2055](https://github.com/microsoft/mssql-jdbc/pull/2055)
+- Fixed for BigDecimal Preciion/Scale [2051](https://github.com/microsoft/mssql-jdbc/pull/2051)
```
Comment (Jeffery-Wasty) [inline CHANGELOG.md:11]: Fixed --> Fix

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,21 @@ All notable changes to this project will be documented in this file.
...
+- Update to check for ExtendedSocketOption class [2043](https://github.com/microsoft/mssql-jdbc/pull/2043)
+- Upgraded dependencies versions [2040](https://github.com/microsoft/mssql-jdbc/pull/2040) [1966](https://github.com/microsoft/mssql-jdbc/pull/1966)
```
Comment (Jeffery-Wasty) [inline CHANGELOG.md:13]: dependencies --> dependency

File: CHANGELOG.md (hunk 1 above)
Comment (David-Engel) [inline CHANGELOG.md:11]:
```suggestion
- Fixed BigDecimal Precision/Scale issue [2051](https://github.com/microsoft/mssql-jdbc/pull/2051)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,21 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 
+## [12.2.0] Preview Release
+### Added
+
+### Changed
+- Renamed driver DefaultAzureCredential authentication to ActiveDirectoryDefault [2055](https://github.com/microsoft/mssql-jdbc/pull/2055)
+- Fixed for BigDecimal Preciion/Scale [2051](https://github.com/microsoft/mssql-jdbc/pull/2051)
+- Update to check for ExtendedSocketOption class [2043](https://github.com/microsoft/mssql-jdbc/pull/2043)
```
Comment (David-Engel) [inline CHANGELOG.md:12]:
```suggestion
- Update to dynamically check for ExtendedSocketOption class [2043](https://github.com/microsoft/mssql-jdbc/pull/2043)
```

File: CHANGELOG.md (hunk 2 above)
Comment (David-Engel) [inline CHANGELOG.md:13]:
```suggestion
- Upgraded dependency versions [2040](https://github.com/microsoft/mssql-jdbc/pull/2040) [1966](https://github.com/microsoft/mssql-jdbc/pull/1966)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,21 @@ All notable changes to this project will be documented in this file.
...
+- Fixed incorrect updateCount [2013](https://github.com/microsoft/mssql-jdbc/pull/2013)
+- Misc code cleanup and test fixes [2053](https://github.com/microsoft/mssql-jdbc/pull/2053) [2049](https://github.com/microsoft/mssql-jdbc/pull/2049) [2048](https://github.com/microsoft/mssql-jdbc/pull/2048) [2044](https://github.com/microsoft/mssql-jdbc/pull/2044) [2041](https://github.com/microsoft/mssql-jdbc/pull/2041) [2031](https://github.com/microsoft/mssql-jdbc/pull/2031) [2029](https://github.com/microsoft/mssql-jdbc/pull2029) [2016](https://github.com/microsoft/mssql-jdbc/pull/2016) [2014](https://github.com/microsoft/mssql-jdbc/pull/2014) [1991](https://github.com/microsoft/mssql-jdbc/pull/1991) [1977](https://github.com/microsoft/mssql-jdbc/pull/1977) [1974](https://github.com/microsoft/mssql-jdbc/pull/1974) [1970](https://github.com/microsoft/mssql-jdbc/pull/1970) [1969](https://github.com/microsoft/mssql-jdbc/pull/1969) [1962](https://github.com/microsoft/mssql-jdbc/pull/1962)
+- Added locks to encrypt/decrypt in truststore password obfuscation [1968](https://github.com/microsoft/mssql-jdbc/pull/1968)
```
Comment (David-Engel) [inline CHANGELOG.md:17]:
```suggestion
- Fixed concurrency issues in encrypt/decrypt obfuscation methods for truststore password [1968](https://github.com/microsoft/mssql-jdbc/pull/1968)
```

**Comment by tkyc** [comment]
These were recently pushed as well. We'll want to add these to the changelog also.

https://github.com/microsoft/mssql-jdbc/pull/1923
https://github.com/microsoft/mssql-jdbc/issues/1590

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by tkyc — APPROVED** [review]

**Pull request merged**